lib/instrumentation/winston.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const createFormatter = require('@newrelic/winston-enricher/lib/createFormatter')
+const API = require('../../api')
+
+module.exports = function instrument(agent, winston, _, shim) {
+  const config = agent.config
+  if (
+    !(
+      config.application_logging &&
+      config.application_logging.enabled &&
+      config.application_logging.local_decorating &&
+      config.application_logging.local_decorating.enabled
+    )
+  ) {
+    shim.logger.debug('Application logging not enabled. Not instrumenting winston...')
+    return // do not instrument logs if logging is not enabled
+  }
+
+  // create an instrumented formatter to enrich and report logs
+  const api = new API(agent)
+  api.shim = shim
+  const instrumentedFormatter = createFormatter(api, winston)
+
+  // wrap logger creation to combine or insert our formatter
+  shim.wrap(winston, 'createLogger', function wrapCreate(shim, createLogger) {
+    return function createWrappedLogger(opts = {}) {
```

Comment:
this is probably ok but we typically just cast the arguments to an array and add more defense and call `.apply` and bind the proper this context to the original function

```js
    return function createWrappedLogger() {
      const args = shim.argsToArray.apply(shim, arguments)
      if (!shim.isObject(args[0])) {
        return createLogger.apply(this, args)
      }

      // combine top-level formatter with our own
      if ('format' in args[0]) {
        args[0].format = winston.format.combine(args[0].format, instrumentedFormatter())
      } else {
        args[0].format = instrumentedFormatter()
      }

      // combine transport-level formatters with our own
      if ('transports' in args[0]) {
        args[0].transports = args[0].transports.map((transport) => {
          if (transport.format) {
            transport.format = winston.format.combine(transport.format, instrumentedFormatter())
          }
          return transport
        })
      }

      return createLogger.apply(this, args)
    }
```

Reply:
Fixed to match your suggestion.